README.md

```diff
@@ -41,31 +41,100 @@ callback parameter like so:
       }
     });
 
-The `eachLine` function also returns an object with one property, `then`.  If a
-callback is provided to `then`, it will be called once all lines have been read.
+You can provide an optional second node-style callback that will be called with
+`(err)` on failure or `()` when finished (even if you manually terminate iteration
+by returning `false` from the iteratee):
 
     var lineReader = require('line-reader');
 
     // read all lines:
     lineReader.eachLine('file.txt', function(line) {
       console.log(line);
-    }).then(function () {
+    }).then(function (err) {
+      if (err) throw err;
       console.log("I'm done!!");
     });
 
-
 For more granular control, `open`, `hasNextLine`, and `nextLine` maybe be used
-to iterate a file:
+to iterate a file (but you must `close` it yourself):
 
     // or read line by line:
-    lineReader.open('file.txt', function(reader) {
+    lineReader.open('file.txt', function(err, reader) {
+      if (err) throw err;
       if (reader.hasNextLine()) {
-        reader.nextLine(function(line) {
-          console.log(line);
+        reader.nextLine(function(err, line) {
+          try {
+            if (err) throw err;
+            console.log(line);
+          } finally {
+            reader.close(function(err) {
+              if (err) throw err;          
+            });
+          }
+        });
+      }
```

Comment:
Please put the closing brace and the `else` on the same line, like so: `} else {` for all changes.